### PR TITLE
[fix] add sum over Poisson.log_prob in loss function

### PR DESCRIPTION
**Pull request by felixzinn**
in the calculation of the poisson term of the likelihood function the sum has to be taken before the constraints are added with `loss_val += evm.util.sum_over_leaves(constraints)` because `Poisson.log_prob` returns a multi-dimensional array with the same shape as `observation`

so it has to be
```python
@eqx.filter_jit
def loss(dynamic_model, static_model, hists, observation):
    model = eqx.combine(dynamic_model, static_model)
    expectations = model(hists)
    constraints = evm.loss.get_log_probs(model)
    loss_val = (
        evm.pdf.Poisson(evm.util.sum_over_leaves(expectations))
        .log_prob(observation)
        .sum()
    )
    # add constraint
    loss_val += evm.util.sum_over_leaves(constraints)
    return -jnp.sum(loss_val)
```

changed in the documentation and examples

### CODE REVIEW

**Comment by pfackeldey** [comment]
Thanks 👍
